connection.go

```diff
@@ -162,28 +165,101 @@ func (mc *mysqlConn) Prepare(query string) (driver.Stmt, error) {
 	return stmt, err
 }
 
+// https://github.com/mysql/mysql-server/blob/mysql-5.7.5/libmysql/libmysql.c#L1150-L1156
+func (mc *mysqlConn) escapeBytes(v []byte) string {
+	var escape func([]byte) []byte
+	if mc.status&statusNoBackslashEscapes == 0 {
+		escape = EscapeString
+	} else {
+		escape = EscapeQuotes
+	}
+	return "'" + string(escape(v)) + "'"
+}
+
+func (mc *mysqlConn) buildQuery(query string, args []driver.Value) (string, error) {
+	chunks := strings.Split(query, "?")
```

Comment:
I'll rewrite it in two-pass way.  First pass is for calculating required length. Second pass is for building `[]bytes`.